README.md

```diff
@@ -51,7 +58,233 @@ custom features to work with multiple nodes and retry logic.
 
 Examples of usage can be found in the [examples](https://github.com/bitcoin-sv/go-broadcast-client/tree/main/examples)
 
-# Transaction Statuses returned by the library
+## ClientBuilder Methods
+
+Client allows you to create your own client with setting it with multiple nodes and custom http client.
+
+### WithArc Method
+
+```go
+// Set the access token and url for the node
+ token := ""
+ apiURL := "https://tapi.taal.com/arc"
+
+ cfg := broadcast_client.ArcClientConfig{
+  Token:  token,
+  APIUrl: apiURL,
+ }
+
+ client := broadcast_client.Builder().
+  WithArc(cfg).
+  Build()
+```
+
+We can also call multiple times the method `WithArc` to set multiple nodes.
+
+```go
+// Set the access token and url for the node
+ client := broadcast_client.Builder().
+  WithArc(cfg1).
+  WithArc(cfg2).
+  WithArc(cfg3).
+  Build()
+```
+
+What is the call order if we have multiple nodes configured?
+
+We use the **strategy** to define the order of the calls. The default strategy is `OneByOne` in RoundRobin algorith that will call the nodes in the order they were set.
+
+### WithHTTPClient Method
+
+```go
+// (...)
+ clent := broadcast_client.Builder().
+  WithArc(cfg).
+  WithHTTPClient(&http.Client{}).
+  Build()
+```
+
+We can use the method `WithHTTPClient` to set the custom http client.
+It needs to implement the interface `HTTPInterface` that is defined in the [httpclient.go](/broadcast/internal/httpclient/http_client.go#L35)
+
+```go
+type HTTPInterface interface {
+ DoRequest(ctx context.Context, pld HTTPRequest) (*http.Response, error)
+}
+```
+
+**Only if all of them fail, we will return the error to the user.**
```

Comment:
I belive this line belongs to the previous paragraph

Reply:
Yes, fixed